### PR TITLE
Remove SVG unnecessary data

### LINKED ISSUE #5: Optimize SVG for web
SVG files have weird data on them, like the path where the file was saved on disk. See https://github.com/readthedocs/site-community/pull/25#discussion_r784184333

### PR DESCRIPTION
**Pull request by nienn**
Removes Inkscape and other unnecessary data from SVG assets.

* Fixes #5 
* Related to https://github.com/readthedocs/site-community/pull/25#discussion_r784184333

### CODE REVIEW

**Review by humitos — APPROVED** [review]
Looks good to me!

For some reason, GitHub shows the new images a little smaller than the old ones, but I don't it's really important since they are svg and scalable without losing quality.

**Review by agjohnson — CHANGES_REQUESTED** [review]
Seeing it now, we probably want a middle ground here between editable sources and ready to host files.

I was originally describing just removing the offending inkscape attributes, but it does make sense to optimize *everything* in the svgs. So :+1: there!

But we should also preserve all of the attributes we'd use in editing, as this repo is also the editable sources for the SVGs, not just the end result images.

Does it make sense to maybe have a `src/` path with the non-optimized images and replace the root path images with optimized versions? If not, we need to preserve the editable formats somewhere else. We could also take this one step further and if we have them anywhere, store the source svgs with objects instead of paths, for easier editing.

**Comment by nienn** [comment]
>For some reason, GitHub shows the new images a little smaller than the old ones

I removed many attributes, several that might be controlling some kind of scaling or positioning so some shift might happen.
It shouldn't be a problem as the final image size should be set by CSS.

>But we should also preserve all of the attributes we'd use in editing, as this repo is also the editable sources for the SVGs, not just the end result images.

The current submitted files are fully editable for me in Illustrator. Haven't tried other editors but should be editable anywhere... are they not?

**Comment by humitos** [comment]
I don't personally think we need "inkscape specific configs" in the SVG file. In case we need to save some metadata (e.g. layers, order of them, etc) we should store software-specific files instead, like `.ai` for illustrator or the inkscape-specific one (which, unfortunately, to add confusion, is also `.svg`).

However, at this point, I don't think this is needed given the simplicity of these files. I'd say that we can store software-specific files in the future if we need to, tho.

**Comment by nienn** [comment]
>  In case we need to save some metadata (e.g. layers, order of them, etc) we should store software-specific files instead

My point is that the current files are already fully layered, vectorised and editable in most vector programs. Am I missing something? Is there a specific information we need to keep?

**Comment by agjohnson** [comment]
Yeah, the original files have all of the extra font information etc from Inkscape. These look to be paths instead of objects however, but this is at least a little better than having none of those attributes ready for editing.

**Comment by nienn** [comment]
I added the original `.svg` files back inside `src/assets/`. Didn't keep the `.png`s inside that same folder thought, as they are generated from the source `.svg`s.

**Comment by agjohnson** [comment]
Cool, that looks better. I also probably have the original svg format around, with editable objects/text instead of paths. I could see if i can dig those up later.

But to go back to Manuel's point, the viewport size looks odd. If I use these optimized versions as a drop in replacement where we have been using these original svg, I'm getting odd behavior.

For example, our docs display this svg as 0px x 0px:

image

The file is loaded correctly by my browser, but is scaled down to 0x0 for some reason. The orignal svg had view port sizing attributes -- width, height -- which were lost here it seems.

**Comment by nienn** [comment]
Yeah, Illustrator has a known problem dealing with `viewBox`. I don't know that there's an easy fix and we can't manually change the values as that will throw all other coordinates off. This shouldn't be causing the problem however, that's more probably due to removing the `width` and `height` values. I added those back to the files `logo-<dark/light>.svg` and `logo-wordmark-<dark/light>.svg`. The other files didn't have it set originally.

I also removed the `xmlns:xlink` attribute, that is required only when using the `xlink:href` attribute, which we do not, and were both [deprecated](https://developer.mozilla.org/en-US/docs/Web/SVG/Attribute/xlink:href) in SVG 2 anyway.

File: src/assets/logo-light.svg [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+   xmlns="http://www.w3.org/2000/svg"
+   xmlns:sodipodi="http://sodipodi.sourceforge.net/DTD/sodipodi-0.dtd"
+   xmlns:inkscape="http://www.inkscape.org/namespaces/inkscape"
+   width="360"
+   height="360"
+   id="svg4006"
+   version="1.1"
+   inkscape:version="0.48.5 r10040"
+   sodipodi:docname="logo-dark.svg"
+   inkscape:export-filename="/Users/anthony/dev/readthedocs/guidelines/assets/logo-dark.png"
```
Comment (agjohnson) [inline src/assets/logo-light.svg:18]: I'm still noticing the file path that we wanted to remove on the source images. I was originally just describing removing this line, not necessarily optimizing the files:

```suggestion
```

I see a few below on this PR in the source assets still as well

**Review by agjohnson — COMMENTED** [review]
The optimized files should hopefully be good now, though I still see the line we wanted to remove in the images files under `src/` still. I would look through the files in `src/` and remove any mentions of paths that would have been from my local machine.

This looks good otherwise :+1:

File: src/assets/logo-light.svg (hunk 1 above)
Comment (nienn) [reply src/assets/logo-light.svg:18]: Just to be sure, these are the original source files that I added back. It's just the `inkscape:export-filename` that needs to be removed?

**Comment by davidfischer** [comment]
Just a quick note that if you save from Inkscape as a "Plain SVG" rather than an "Inkscape SVG", all these attributes will be removed. You'll still have individual SVG groups (layers) and everything but you just lose a few Inkscape conveniences when the window opens like the default zoom for this document.

image

**Comment by agjohnson** [comment]
That's handy, easy enough. Inkscape is also scriptable, which makes SVG processing an automated process too. We'll probably land here in the future just so that we have a reproducible process -- I can't optimize future images with Illustrator, but Inkscape works fine for this too.

I'm fine with the source images having the extra attributes, but it's a bit extra silly that local filenames in inkscape get hardcoded into the SVG source. :shrug:

File: src/assets/logo-light.svg (hunk 1 above)
Comment (agjohnson) [reply src/assets/logo-light.svg]: Yeah, that was the original issue that was raised

https://github.com/readthedocs/site-community/pull/25#discussion_r784184333

I think this got misinterpreted as "optimize" somewhere along the way.

**Review by agjohnson — APPROVED** [review]

**Comment by pradyunsg** [comment]
Could the links for the assets on https://read-the-docs-guidelines.readthedocs-hosted.com/branding.html#logo-guidelines be changed to optimised ones?

**Comment by nienn** [comment]
The links are there. Changes still need to be deployed. The latest revision online is [3fc8648](https://github.com/readthedocs/guidelines/commit/3fc8648527f266b0fa02baefb333fcc451227eec) from July 26, 2021.

**Pull request merged**